target_chains/sui/contracts/sources/pyth.move

```diff
@@ -651,12 +776,329 @@ module pyth::pyth_tests{
+    #[test]
+    fun test_create_and_update_single_price_feed_with_accumulator_success() {
+        use pyth::data_source::Self;
+
+        let base_update_fee = 50;
+        let coins_to_mint = 5000;
+        let emitter_address = x"71f8dcb863d176e2c420ad6610cf687359612b6fb392e0642b0ca6b1f186aa3b";
+        let initial_guardians = vector[x"7E5F4552091A69125d5DfCb7b8C2659029395Bdf"];
+        let data_sources = vector[data_source::new(1, external_address::new(bytes32::from_bytes(x"71f8dcb863d176e2c420ad6610cf687359612b6fb392e0642b0ca6b1f186aa3b")))];
+        //new(emitter_chain: u64, emitter_address: ExternalAddress)
+        let (scenario, coins, clock) = setup_test(500, 23, emitter_address, data_sources, initial_guardians, base_update_fee, coins_to_mint);
+
+        test_scenario::next_tx(&mut scenario, DEPLOYER);
+
+        let pyth_state = take_shared<PythState>(&scenario);
+        let worm_state = take_shared<WormState>(&scenario);
+
+        let verified_vaa = get_verified_vaa_from_accumulator_message(&worm_state, TEST_ACCUMULATOR_SINGLE_FEED, &clock);
+
+        test_scenario::next_tx(&mut scenario, DEPLOYER);
+
+        pyth::create_price_feeds_using_accumulator(
+            &mut pyth_state,
+            TEST_ACCUMULATOR_SINGLE_FEED,
+            verified_vaa,
+            &clock,
+            ctx(&mut scenario)
+        );
+
+        // Affirm that 1 object, which correspond to the 1 new price info object
+        // containing the price feeds were created and shared.
+        let effects = test_scenario::next_tx(&mut scenario, DEPLOYER);
+        let shared_ids = test_scenario::shared(&effects);
+        let created_ids = test_scenario::created(&effects);
+        assert!(vector::length<ID>(&shared_ids)==1, 0);
+        assert!(vector::length<ID>(&created_ids)==1, 0);
+
+        let price_info_object_1 = take_shared<PriceInfoObject>(&scenario);
+
+        // Create authenticated price infos
+        verified_vaa = get_verified_vaa_from_accumulator_message(&worm_state, TEST_ACCUMULATOR_SINGLE_FEED, &clock);
+        let auth_price_infos = pyth::create_authenticated_price_infos_using_accumulator(
+            &pyth_state,
+            TEST_ACCUMULATOR_SINGLE_FEED,
+            verified_vaa,
+            &clock
+        );
+
+        test_scenario::next_tx(&mut scenario, DEPLOYER);
+        auth_price_infos = update_single_price_feed(
+            &mut pyth_state,
+            auth_price_infos,
+            &mut price_info_object_1,
+            coins,
+            &clock
+        );
+
+        test_scenario::next_tx(&mut scenario, DEPLOYER);
+        authenticated_vector::destroy<PriceInfo>(auth_price_infos);
+
```

Comment:
shouldn't there be some assertion here or something that the updates happened correctly?